tutorials/Machine_Translation_using_PyTorch_Ignite.ipynb

```diff
@@ -0,0 +1,2295 @@
+   },
+   "source": [
+    "train_data = TransformerDataset(\n",
+    "    train_dataset,\n",
+    "    dataset_configs[\"source_language\"],\n",
+    "    dataset_configs[\"target_language\"],\n",
+    "    tokenizer,\n",
+    "    dataset_configs[\"max_length\"],\n",
+    "    dataset_configs[\"train_dataset_length\"],\n",
+    ")\n",
+    "val_data = TransformerDataset(\n",
+    "    validation_dataset,\n",
+    "    dataset_configs[\"source_language\"],\n",
+    "    dataset_configs[\"target_language\"],\n",
+    "    tokenizer,\n",
+    "    dataset_configs[\"max_length\"],\n",
+    "    dataset_configs[\"validation_dataset_length\"],\n",
+    ")\n"
+   ],
+   "execution_count": 10,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "7C19xqP_h9gj"
+   },
+   "source": [
+    "## Trainer\n",
+    "The trainer takes a batch of input and passes them through the model (along with targets in this case) and gets the loss.\n",
+    "\n",
+    "#### Mixed Precision\n",
+    "The forward pass is wrapped in the autocast context manager for mixed precision training. It's turned off in this example as there won't be any memory advantages with ``batch_size`` 1 or 2. Change the ``with_amp`` flag in config to turn it on.\n",
+    "\n",
+    "#### Gradient Accumulation\n",
+    "Gradient accumulation is implemented as batch size of 1 would lead to noisy updates otherwise. Check the ``accumulation_steps`` variable in config to define the number of steps to accumulate the gradient. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {
+    "id": "hTIlKn9CJGMC"
+   },
+   "source": [
+    "# Create Trainer\n",
+    "def create_trainer(model, optimizer, with_amp, train_sampler, logger):\n",
+    "    device = idist.device()\n",
+    "    scaler = GradScaler(enabled=with_amp)\n",
+    "\n",
+    "    def train_step(engine, batch):\n",
+    "        model.train()\n",
+    "\n",
+    "        if batch[\"tgt\"].device != device:\n",
+    "            batch = {\n",
+    "                k: v.to(device, non_blocking=True, dtype=torch.long)\n",
+    "                for (k, v) in batch.items()\n",
+    "            }\n",
+    "\n",
+    "        src_ids = batch[\"input_ids\"]\n",
+    "        src_attention_mask = batch[\"attention_mask\"]\n",
+    "        tgt = batch[\"tgt\"]\n",
+    "\n",
+    "        with autocast(enabled=with_amp):\n",
+    "            y = model(src_ids, src_attention_mask, labels=tgt)\n",
+    "            loss = y[\"loss\"]\n",
+    "            loss /= config[\"accumulation_steps\"]\n",
+    "\n",
+    "        scaler.scale(loss).backward()\n",
+    "\n",
+    "        if engine.state.iteration % config[\"accumulation_steps\"] == 0:\n",
+    "            scaler.step(optimizer)\n",
+    "            scaler.update()\n",
+    "            optimizer.zero_grad()\n",
+    "\n",
+    "        return {\"batch loss\": loss.item()}\n",
+    "\n",
+    "    trainer = Engine(train_step)\n",
```

Comment:
Maybe it could worth to split this and add a few words about `common.setup_common_training_handlers` ?